router/client/src/lib.rs

```diff
@@ -44,3 +48,37 @@ impl From<transport::Error> for ClientError {
 }
 
 pub type Result<T> = std::result::Result<T, ClientError>;
+
+// Small conversion so that we can use `vec![Chunk::..., ...]` internally
+// rather than `vec![InputChunk { chunk: Some(Chunk::...)}, ...]`.
+impl From<Vec<Chunk>> for Input {
+    fn from(chunks: Vec<Chunk>) -> Self {
+        Input {
+            chunks: chunks
+                .into_iter()
+                .map(|c| InputChunk { chunk: Some(c) })
+                .collect(),
+        }
+    }
+}
```

Comment:
This means an entire realloc of the full vec most of the time.

I think we can handle ergonomics a bit better than havnig to realloc all the time, no ?